XenAdmin/Dialogs/ScheduledSnapshots/ScheduledSnapshotsDialog.cs

```diff
@@ -144,90 +185,130 @@ private void RefreshPoolTitle(Pool pool)
                         protectedVMs++;
                 }
             }
-            this.Text = Messages.VMSS_DIALOG_TITLE;
-            labelTopBlurb.Text = Messages.VMSS_DIALOG_TEXT;
+
             labelPolicyTitle.Text = string.Format(Helpers.IsPool(pool.Connection)
                                                         ? Messages.VMSS_SCHEDULED_SNAPSHOTS_DEFINED_FOR_POOL
                                                         : Messages.VMSS_SCHEDULED_SNAPSHOTS_DEFINED_FOR_SERVER,
                                                     pool.Name().Ellipsise(45), protectedVMs, realVMs);
         }
 
-        void VMSSCollectionChanged(object sender, EventArgs e)
+        private void VMSSCollectionChanged(object sender, EventArgs e)
         {
-            LoadPolicies();
+            if (!bgWorker.IsBusy)

```

Comment:
I think we could use a QueuedBackgroundWorker here, so that we don't ignore CollectionChanged events that happen while the bgWorker is busy.